contracts/cw-ics721-bridge/src/ibc.rs

```diff
@@ -0,0 +1,413 @@
+        // For each tokenID:
+        //   1. Get the escrow address for this destination port and
+        //      channel.
+        //   2. Get the cw721 address for this classID.
+        //   3. Transfer the tokenID from escrow to receiver.
+        //
+        // See `execute_receive_nft` for a description of why this works.
+
+        // The escrow protects us from overwithdrawal by an evil chain
+        // because the cw721s underlying all this will start failing
+        // the transactions as the escrow for the channel will run out
+        // of NFTs. If we held all these in the bridge contract a
+        // malicious chain could connect to us using the same source
+        // prefix as a different chain and drain all of its funds.
+        //
+        // TL;DR - Don't get rid of the escrows. Make sure to use
+        // `packet.dest` in the channel field below.
+
+        // Transfer all of the requested NFTs out of the channel and
+        // to the receiver.
+        let message = ExecuteMsg::BatchTransferFromChannel {
+            channel: packet.dest.channel_id,
+            class_id: class_id.to_string(),
+            token_ids: data.token_ids,
+            receiver: data.receiver,
+        };
+
+        let message = WasmMsg::Execute {
+            contract_addr: env.contract.address.into_string(),
+            msg: to_binary(&message)?,
+            funds: vec![],
+        };
+        let message = SubMsg::reply_always(message, BATCH_TRANSFER_FROM_CHANNEL_REPLY_ID);
+
+        Ok(IbcReceiveResponse::default()
+            .add_attribute("method", "ics721_transfer_source")
+            .add_submessage(message))
+    } else {
+        // The token is being sent to this chain from another
+        // one. Push to classID and dispatch submessage to create new x
+        // cw721 (if needed) and mint for the receiver.
+        let local_prefix = get_endpoint_prefix(&packet.dest);
+        let local_class_id = format!("{}{}", local_prefix, data.class_id);
+
+        // We can not dispatch multiple submessages and still handle
+        // errors and rollbacks correctly [1]. As such, we bundle
+        // these steps into one message that is only callable by the
+        // contract itself.
+        //
+        // [1] https://github.com/CosmWasm/cosmwasm/blob/main/IBC.md#acknowledging-errors
+        let ibc_ack_check_message = SubMsg::reply_always(
+            WasmMsg::Execute {
+                contract_addr: env.contract.address.into_string(),
+                msg: to_binary(&ExecuteMsg::DoInstantiateAndMint {
+                    class_id: local_class_id,
+                    class_uri: data.class_uri,
+                    token_ids: data.token_ids,
+                    token_uris: data.token_uris,
+                    // FIXME: ics20 seems to set the receiver field as a
+                    // bech32 address. IF we need to do this, need to convert
+                    // first.
+                    receiver: data.receiver,
```

Comment:
Here, we want the sender of the transaction (or the owner of the NFT) to receive the NFT on the receiving chain. 
`receive: data.receiver` should be changed to `receiver: data.sender`.
Currently, the token gets minted to the receiving bridge contract